dbt/adapters/postgres.py

```diff
@@ -165,3 +167,64 @@ def cancel_connection(cls, profile, connection):
         res = cursor.fetchone()
 
         logger.debug("Cancel query '{}': {}".format(connection_name, res))
+
+    @classmethod
+    def convert_text_type(cls, agate_table, col_idx):
+        return "text"
```

Comment:
why go with this approach here, instead of getting the max length using http://agate.readthedocs.io/en/1.6.0/api/aggregations.html#agate.MaxLength (similar to `convert_number_type`)?

Reply:
PostgreSQL and Snowflake both treat the N portion of `varchar(N)` as the number of characters whereas in Redshift, N is the number of bytes. I could use the same logic and the N in PG / Snowflake would be 3-4x higher than necessary, which isn't a big deal. But I figured since there is no performance difference with using "text" I would just isolate where finding the max length happens.